code/datums/api_handler.dm

```diff
@@ -49,9 +49,9 @@ var/global/datum/apiHandler/apiHandler
 	 * @return (list|boolean) list containing parsed data response from api, 1 if forceResponse is false
 	 *
 	 */
-	proc/queryAPI(route = "", query = list(), forceResponse = 0, attempt = 1)
+	proc/queryAPI(route = "", query = list(), forceResponse = 0, attempt = 1, forceErrorException = 0)
 		if (!enabled || !route)
-			src.apiError("API Error: Cancelled query due to [!enabled ? "disabled apiHandler" : "missing route parameter"]")
+			src.apiError("API Error: Cancelled query due to [!enabled ? "disabled apiHandler" : "missing route parameter"]", forceErrorException)
```

Comment:
shouldn't you be setting the exception to a value?

Reply:
That logic happens in `apiHandler.apiError`; the `forceErrorException` argument just allows a query to bypass a check in that proc which prevents throwing the exception if the server is being run locally.

Reply:
yes i know, but you're not setting it to anything, which might cause it to be interpreted as the 2nd argument I think

Reply:
I think that's alright here? The second argument of `apiHandler.apiError` is `forceErrorException`. I could make it explicit (`forceErrorException = forceErrorException`) although the way it is written here works.

Reply:
bleh, i was getting it confused with proc/queryAPI, ignore me